internal/pkg/service/stream/api/service/aggregation.go

```diff
@@ -3,11 +3,59 @@ package service
 import (
 	"context"
 
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"golang.org/x/exp/maps"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/api/gen/stream"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
 )
 
 func (s *service) AggregateSources(ctx context.Context, d dependencies.BranchRequestScope, payload *stream.AggregateSourcesPayload) (res *stream.AggregationSourcesResult, err error) {
-	return nil, errors.NewNotImplementedError()
+	list := func(opts ...iterator.Option) iterator.DefinitionT[definition.Source] {
+		return s.definition.Source().List(d.BranchKey(), opts...)
+	}
+
+	response, err := s.mapper.NewAggregationSourcesResponse(ctx, d.BranchKey(), payload.SinceID, payload.Limit, list)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.addSinksToAggregationResponse(ctx, d, response)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, err
+}
+
+func (s *service) addSinksToAggregationResponse(ctx context.Context, d dependencies.BranchRequestScope, response *stream.AggregationSourcesResult) error {
+	sourceKeys := make(map[key.SourceKey]int)
+	for i, source := range response.Sources {
+		sourceKey := key.SourceKey{
+			BranchKey: d.BranchKey(),
+			SourceID:  source.SourceID,
+		}
+		sourceKeys[sourceKey] = i
+	}
```

Comment:
So we create map of `sourceKeys` to deduplicate it and then we cast it into slice `[]key.SourceKey`. 
Two things

1. Is it possible to write somewhere that we need to deduplicate the entries? Or maybe move it to the top level layer so we work only with deduplicated entries.
2. Do we need to cast `map` into `slice` ? It need to loop over fields to actually create slice. Cannot we use the keys in the map? The loop in `GetSourcesWithSinksAndStatistics` will be change into `for key, _ := range sourceKeys`
https://github.com/keboola/keboola-as-code/pull/1819/commits/266712c97ba90386318011029620a74eea733223#diff-8c5b3a594c9443cdd704ef50193f2fd6f6c9bb148d52fe716c85a6db31da8df6R99

we will reduce numerous casting `map -> slice -> loop` into `map -> loop`. This would have effect when we would like to preserve the values and the `GetSorucesWithSinksAndStatistics` would have side effect on `sourceKeys`